frontend/lib/types/CompositeType.cpp

```diff
@@ -96,21 +96,38 @@ void CompositeType::stringify(std::ostream& ss,
 
   auto sorted = sortedSubstitutions();
 
-  if (superType || !sorted.empty()) {
+  bool printSupertype =
+    superType != nullptr && stringKind != StringifyKind::CHPL_SYNTAX;
+
+  if (printSupertype || !sorted.empty()) {
     bool emittedField = false;
     ss << "(";
 
-    if (superType != nullptr && stringKind != StringifyKind::CHPL_SYNTAX) {
+    if (printSupertype) {
       ss << "super:";
       superType->stringify(ss, stringKind);
       emittedField = true;
     }
 
     for (const auto& sub : sorted) {
       if (emittedField) ss << ", ";
-      sub.first.stringify(ss, stringKind);
-      ss << ":";
-      sub.second.stringify(ss, stringKind);
+
+      if (stringKind != StringifyKind::CHPL_SYNTAX) {
+        sub.first.stringify(ss, stringKind);
+        ss << ":";
+        sub.second.stringify(ss, stringKind);
+      } else {
```

Comment:
There seems to be an implicit link here between CHPL_SYNTAX and the substitutions. If that's accurate, can you add a comment? Or is there a way of writing this that makes the relationship clearer?

Reply:
I don't think that's true. Substitutions are always printed, but without CHPL_SYNTAX, we print the IDs and qualified types; with CHPL_SYNTAX, we only print (unqualified) types, or the param values. This matches the actual Chapel syntax, `myType(int, 42)`.

Reply:
I think your reply to the first comment clears this one up 👍 . I think I misread some of the boolean logic and got caught up in thinking there was some gap that doesn't exist.